code/modules/mod/modules/modules_timeline.dm

```diff
@@ -0,0 +1,336 @@
+	icon_state = "rewinder"
+	module_type = MODULE_USABLE
+	complexity = 1
+	use_power_cost = DEFAULT_CELL_DRAIN * 5
+	incompatible_modules = list(/obj/item/mod/module/rewinder)
+	cooldown_time = 30 SECONDS
+	removable = FALSE //copy paste this comment - no timeline modules should be removable
+
+/obj/item/mod/module/rewinder/on_use()
+	. = ..()
+	if(!.)
+		return
+	playsound(src, 'sound/items/modsuits/time_anchor_set.ogg', 50, TRUE)
+	mod.wearer.AddComponent(/datum/component/dejavu/timeline, 1, 10 SECONDS)
+	RegisterSignal(mod, COMSIG_MOD_ACTIVATE, .proc/on_activate_block)
+	addtimer(CALLBACK(src, .proc/unblock_suit_activation), 10 SECONDS)
+
+///unregisters the modsuit deactivation blocking signal, after dejavu functionality finishes.
+/obj/item/mod/module/rewinder/proc/unblock_suit_activation()
+	UnregisterSignal(mod, COMSIG_MOD_ACTIVATE)
+
+///Signal fired when wearer attempts to activate/deactivate suits
+/obj/item/mod/module/rewinder/proc/on_activate_block(datum/source, user)
+	SIGNAL_HANDLER
+	balloon_alert(user, "not while rewinding!")
+	return MOD_CANCEL_ACTIVATE
+
+///timestopper - Need I really explain? It's the wizard's time stop, but the user channels it by not moving instead of a duration.
+/obj/item/mod/module/timestopper
+	name = "MOD timestopper module"
+	desc = "A module that can halt time in a small radius around the user... for as long as they \
+	want! Great for monologues or lunch breaks. Keep in mind moving will end the stop, and the \
+	module has a hefty cooldown period to avoid reality errors."
+	icon_state = "timestop"
+	module_type = MODULE_USABLE
+	complexity = 3
+	use_power_cost = DEFAULT_CELL_DRAIN * 5
+	incompatible_modules = list(/obj/item/mod/module/timestopper)
+	cooldown_time = 60 SECONDS
+	removable = FALSE //copy paste this comment - no timeline modules should be removable
+	///the current timestop in progress
+	var/obj/effect/timestop/channelled/timestop
+
+/obj/item/mod/module/timestopper/on_use()
+	. = ..()
+	if(!.)
+		return
+	if(timestop)
+		mod.balloon_alert(mod.wearer, "already freezing time!")
+		return
+	RegisterSignal(mod, COMSIG_MOD_ACTIVATE, .proc/on_activate_block)
+	timestop = new /obj/effect/timestop/channelled(get_turf(mod.wearer), 2, INFINITY, list(mod.wearer))
+	RegisterSignal(timestop, COMSIG_PARENT_QDELETING, .proc/unblock_suit_activation)
+
+///unregisters the modsuit deactivation blocking signal, after timestop functionality finishes.
+/obj/item/mod/module/timestopper/proc/unblock_suit_activation(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(mod, COMSIG_MOD_ACTIVATE)
+	on_deactivation()
+
+///Signal fired when wearer attempts to activate/deactivate suits
+/obj/item/mod/module/timestopper/proc/on_activate_block(datum/source, user)
+	SIGNAL_HANDLER
+	balloon_alert(user, "not while channelling timestop!")
+	return MOD_CANCEL_ACTIVATE
+
+///TEM - Lets you eradicate people.
+/obj/item/mod/module/tem
```

Comment:
the gun you take out of the suit looked cool are we sure we dont want to make this a device module

Reply:
(or add a worn overlay that looks similar to the gun)

Reply:
the old one was kinda a pos, i can make this work the same